diff --git a/BUILD.bazel b/BUILD.bazel
index 577aeaf..de6df58 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -23,4 +23,7 @@ package(
 exports_files(["LICENSE"])
 
 # gazelle:prefix github.com/openconfig/gnoi
+# gazelle:proto_import_prefix github.com/openconfig/gnoi
+# gazelle:resolve proto proto google/rpc/status.proto @googleapis//google/rpc:status_proto
+# gazelle:resolve proto go google/rpc/status.proto @org_golang_google_genproto_googleapis_rpc//status
 gazelle(name = "gazelle")
diff --git a/bgp/BUILD.bazel b/bgp/BUILD.bazel
index 25118f8..5476ce6 100644
--- a/bgp/BUILD.bazel
+++ b/bgp/BUILD.bazel
@@ -29,6 +29,7 @@ package(
 proto_library(
     name = "bgp_proto",
     srcs = ["bgp.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = [
         "//types:types_proto",
     ],
@@ -51,5 +52,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/bgp",
     proto = ":bgp_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "bgp",
+    embed = [":bgp_go_proto"],
+    importpath = "github.com/openconfig/gnoi/bgp",
 )
diff --git a/bgp/bgp.proto b/bgp/bgp.proto
index 5fe7834..8f8a8c7 100644
--- a/bgp/bgp.proto
+++ b/bgp/bgp.proto
@@ -18,7 +18,7 @@ syntax = "proto3";
 
 package gnoi.bgp;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/bgp";
 
diff --git a/cert/BUILD.bazel b/cert/BUILD.bazel
index 5c6eb48..4e05080 100644
--- a/cert/BUILD.bazel
+++ b/cert/BUILD.bazel
@@ -28,6 +28,7 @@ package(
 proto_library(
     name = "cert_proto",
     srcs = ["cert.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = ["//types:types_proto"],
 )
 
@@ -48,5 +49,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/cert",
     proto = ":cert_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "cert",
+    embed = [":cert_go_proto"],
+    importpath = "github.com/openconfig/gnoi/cert",
 )
diff --git a/cert/cert.proto b/cert/cert.proto
index ba9d9f1..952a700 100644
--- a/cert/cert.proto
+++ b/cert/cert.proto
@@ -20,7 +20,7 @@ syntax = "proto3";
 
 package gnoi.certificate;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/cert";
 
diff --git a/common/BUILD.bazel b/common/BUILD.bazel
index 9b79c47..262880a 100644
--- a/common/BUILD.bazel
+++ b/common/BUILD.bazel
@@ -29,6 +29,7 @@ package(
 proto_library(
     name = "common_proto",
     srcs = ["common.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = ["//types:types_proto"],
 )
 
@@ -41,5 +42,11 @@ go_proto_library(
     name = "common_go_proto",
     importpath = "github.com/openconfig/gnoi/common",
     proto = ":common_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "common",
+    embed = [":common_go_proto"],
+    importpath = "github.com/openconfig/gnoi/common",
 )
diff --git a/common/common.proto b/common/common.proto
index 913c9fc..c62f37a 100644
--- a/common/common.proto
+++ b/common/common.proto
@@ -18,7 +18,7 @@ syntax = "proto3";
 
 package gnoi.common;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/common";
 
diff --git a/containerz/BUILD.bazel b/containerz/BUILD.bazel
index ca7168e..4474db5 100644
--- a/containerz/BUILD.bazel
+++ b/containerz/BUILD.bazel
@@ -27,10 +27,11 @@ package(
 proto_library(
     name = "containerz_proto",
     srcs = ["containerz.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = [
-        "@go_googleapis//google/rpc:status_proto",
+        "//common:common_proto",
         "//types:types_proto",
-        "//common:common_proto"
+        "@googleapis//google/rpc:status_proto",
     ],
 )
 
@@ -52,8 +53,14 @@ go_proto_library(
     importpath = "github.com/openconfig/gnoi/containerz",
     proto = ":containerz_proto",
     deps = [
-        "@go_googleapis//google/rpc:status_go_proto",
-        "//types:types_go_proto",
-        "//common:common_go_proto"
+        "//common",
+        "//types",
+        "@org_golang_google_genproto_googleapis_rpc//status",
     ],
 )
+
+go_library(
+    name = "containerz",
+    embed = [":containerz_go_proto"],
+    importpath = "github.com/openconfig/gnoi/containerz",
+)
diff --git a/containerz/containerz.proto b/containerz/containerz.proto
index 0539a5b..7098670 100644
--- a/containerz/containerz.proto
+++ b/containerz/containerz.proto
@@ -21,9 +21,9 @@ syntax = "proto3";
 
 package gnoi.containerz;
 
-import "common/common.proto";
+import "github.com/openconfig/gnoi/common/common.proto";
 import "google/rpc/status.proto";
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/containerz";
 option (types.gnoi_version) = "0.1.1";
diff --git a/diag/BUILD.bazel b/diag/BUILD.bazel
index 074348d..82020f7 100644
--- a/diag/BUILD.bazel
+++ b/diag/BUILD.bazel
@@ -29,6 +29,7 @@ package(
 proto_library(
     name = "diag_proto",
     srcs = ["diag.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = [
         "//types:types_proto",
     ],
@@ -51,5 +52,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/diag",
     proto = ":diag_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "diag",
+    embed = [":diag_go_proto"],
+    importpath = "github.com/openconfig/gnoi/diag",
 )
diff --git a/diag/diag.proto b/diag/diag.proto
index 1876bf5..5ca592f 100644
--- a/diag/diag.proto
+++ b/diag/diag.proto
@@ -20,7 +20,7 @@ syntax = "proto3";
 
 package gnoi.diag;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/diag";
 
diff --git a/factory_reset/BUILD.bazel b/factory_reset/BUILD.bazel
index 13fabb4..9de31ac 100644
--- a/factory_reset/BUILD.bazel
+++ b/factory_reset/BUILD.bazel
@@ -28,6 +28,7 @@ package(
 proto_library(
     name = "factory_reset_proto",
     srcs = ["factory_reset.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = [
         "//types:types_proto",
     ],
@@ -50,5 +51,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/factory_reset",
     proto = ":factory_reset_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "factory_reset",
+    embed = [":factory_reset_go_proto"],
+    importpath = "github.com/openconfig/gnoi/factory_reset",
 )
diff --git a/factory_reset/factory_reset.proto b/factory_reset/factory_reset.proto
index e8e4fcc..a035c63 100644
--- a/factory_reset/factory_reset.proto
+++ b/factory_reset/factory_reset.proto
@@ -3,7 +3,7 @@ syntax = "proto3";
 
 package gnoi.factory_reset;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/factory_reset";
 
diff --git a/file/BUILD.bazel b/file/BUILD.bazel
index 5349b82..943e675 100644
--- a/file/BUILD.bazel
+++ b/file/BUILD.bazel
@@ -28,6 +28,7 @@ package(
 proto_library(
     name = "file_proto",
     srcs = ["file.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = [
         "//common:common_proto",
         "//types:types_proto",
@@ -52,8 +53,8 @@ go_proto_library(
     importpath = "github.com/openconfig/gnoi/file",
     proto = ":file_proto",
     deps = [
-        "//common:common_go_proto",
-        "//types:types_go_proto",
+        "//common",
+        "//types",
     ],
 )
 
diff --git a/file/file.proto b/file/file.proto
index b5d64d1..16c975e 100644
--- a/file/file.proto
+++ b/file/file.proto
@@ -18,8 +18,8 @@ syntax = "proto3";
 
 package gnoi.file;
 
-import "common/common.proto";
-import "types/types.proto";
+import "github.com/openconfig/gnoi/common/common.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/file";
 
diff --git a/healthz/BUILD.bazel b/healthz/BUILD.bazel
index 039f3b5..248fa20 100644
--- a/healthz/BUILD.bazel
+++ b/healthz/BUILD.bazel
@@ -16,7 +16,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_library")
 # limitations under the License.
 #
 load("@com_github_grpc_grpc//bazel:cc_grpc_library.bzl", "cc_grpc_library")
-load("@rules_proto_grpc//go:defs.bzl", "go_grpc_library", "go_proto_library")
+load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
 
 package(
     default_visibility = ["//visibility:public"],
@@ -26,9 +26,9 @@ package(
 proto_library(
     name = "healthz_proto",
     srcs = ["healthz.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = [
         "//types:types_proto",
-        "//common:common_proto",
         "@com_google_protobuf//:any_proto",
         "@com_google_protobuf//:timestamp_proto",
     ],
@@ -38,10 +38,14 @@ go_proto_library(
     name = "healthz_go_proto",
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/healthz",
-    protos = [":healthz_proto"],
-    deps = [
-        "//types:types_go_proto",
-    ],
+    proto = ":healthz_proto",
+    deps = ["//types"],
+)
+
+go_library(
+    name = "healthz",
+    embed = [":healthz_go_proto"],
+    importpath = "github.com/openconfig/gnoi/healthz",
 )
 
 cc_proto_library(
diff --git a/healthz/healthz.proto b/healthz/healthz.proto
index 7800063..5b4e296 100644
--- a/healthz/healthz.proto
+++ b/healthz/healthz.proto
@@ -26,7 +26,7 @@ package gnoi.healthz;
 
 import "google/protobuf/any.proto";
 import "google/protobuf/timestamp.proto";
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/healthz";
 
diff --git a/layer2/BUILD.bazel b/layer2/BUILD.bazel
index 63f5209..ad80e5d 100644
--- a/layer2/BUILD.bazel
+++ b/layer2/BUILD.bazel
@@ -28,6 +28,7 @@ package(
 proto_library(
     name = "layer2_proto",
     srcs = ["layer2.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = ["//types:types_proto"],
 )
 
@@ -48,5 +49,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/layer2",
     proto = ":layer2_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "layer2",
+    embed = [":layer2_go_proto"],
+    importpath = "github.com/openconfig/gnoi/layer2",
 )
diff --git a/layer2/layer2.proto b/layer2/layer2.proto
index fa90f61..552ac29 100644
--- a/layer2/layer2.proto
+++ b/layer2/layer2.proto
@@ -18,7 +18,7 @@ syntax = "proto3";
 
 package gnoi.layer2;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/layer2";
 
diff --git a/mpls/BUILD.bazel b/mpls/BUILD.bazel
index adf6daa..0aa7874 100644
--- a/mpls/BUILD.bazel
+++ b/mpls/BUILD.bazel
@@ -28,6 +28,7 @@ package(
 proto_library(
     name = "mpls_proto",
     srcs = ["mpls.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = ["//types:types_proto"],
 )
 
@@ -48,5 +49,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/mpls",
     proto = ":mpls_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "mpls",
+    embed = [":mpls_go_proto"],
+    importpath = "github.com/openconfig/gnoi/mpls",
 )
diff --git a/mpls/mpls.proto b/mpls/mpls.proto
index a96ba9b..0c3b308 100644
--- a/mpls/mpls.proto
+++ b/mpls/mpls.proto
@@ -18,7 +18,7 @@ syntax = "proto3";
 
 package gnoi.mpls;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/mpls";
 
diff --git a/os/BUILD.bazel b/os/BUILD.bazel
index eef8825..bc239aa 100644
--- a/os/BUILD.bazel
+++ b/os/BUILD.bazel
@@ -29,6 +29,7 @@ package(
 proto_library(
     name = "os_proto",
     srcs = ["os.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = ["//types:types_proto"],
 )
 
@@ -49,5 +50,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/os",
     proto = ":os_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "os",
+    embed = [":os_go_proto"],
+    importpath = "github.com/openconfig/gnoi/os",
 )
diff --git a/os/os.proto b/os/os.proto
index c9a6410..2709725 100644
--- a/os/os.proto
+++ b/os/os.proto
@@ -3,7 +3,7 @@ syntax = "proto3";
 
 package gnoi.os;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/os";
 
diff --git a/otdr/BUILD.bazel b/otdr/BUILD.bazel
index 5fc1a8a..e5adc38 100644
--- a/otdr/BUILD.bazel
+++ b/otdr/BUILD.bazel
@@ -29,6 +29,7 @@ package(
 proto_library(
     name = "otdr_proto",
     srcs = ["otdr.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = ["//types:types_proto"],
 )
 
@@ -49,5 +50,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/otdr",
     proto = ":otdr_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "otdr",
+    embed = [":otdr_go_proto"],
+    importpath = "github.com/openconfig/gnoi/otdr",
 )
diff --git a/otdr/otdr.proto b/otdr/otdr.proto
index 459deb3..f05cc21 100644
--- a/otdr/otdr.proto
+++ b/otdr/otdr.proto
@@ -18,7 +18,7 @@ syntax = "proto3";
 
 package gnoi.optical;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/otdr";
 
diff --git a/packet_capture/packet_capture.proto b/packet_capture/packet_capture.proto
index 29354c2..2da334d 100644
--- a/packet_capture/packet_capture.proto
+++ b/packet_capture/packet_capture.proto
@@ -2,7 +2,7 @@ syntax = "proto3";
 
 package gnoi.pcap;
 
-option go_package = "github.com/openconfig/gnoi/pcap";
+option go_package = "github.com/openconfig/gnoi/packet_capture;pcap";
 
 // The PacketCapture service exported by targets.
 service PacketCapture {
diff --git a/packet_link_qualification/BUILD.bazel b/packet_link_qualification/BUILD.bazel
index 249bc3a..3b44a38 100644
--- a/packet_link_qualification/BUILD.bazel
+++ b/packet_link_qualification/BUILD.bazel
@@ -5,23 +5,31 @@ load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
 proto_library(
     name = "linkqual_proto",
     srcs = ["packet_link_qualification.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     visibility = ["//visibility:public"],
     deps = [
         "//types:types_proto",
         "@com_google_protobuf//:duration_proto",
         "@com_google_protobuf//:timestamp_proto",
-        "@go_googleapis//google/rpc:status_proto",
+        "@googleapis//google/rpc:status_proto",
     ],
 )
 
 go_proto_library(
     name = "linkqual_go_proto",
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
-    importpath = "github.com/openconfig/gnoi/linkqual",
+    importpath = "github.com/openconfig/gnoi/packet_link_qualification",
     proto = ":linkqual_proto",
     visibility = ["//visibility:public"],
     deps = [
-        "//types:types_go_proto",
-        "@go_googleapis//google/rpc:status_go_proto",
+        "//types",
+        "@org_golang_google_genproto_googleapis_rpc//status",
     ],
 )
+
+go_library(
+    name = "packet_link_qualification",
+    embed = [":linkqual_go_proto"],
+    importpath = "github.com/openconfig/gnoi/packet_link_qualification",
+    visibility = ["//visibility:public"],
+)
diff --git a/packet_link_qualification/packet_link_qualification.proto b/packet_link_qualification/packet_link_qualification.proto
index 8687642..43c27cb 100644
--- a/packet_link_qualification/packet_link_qualification.proto
+++ b/packet_link_qualification/packet_link_qualification.proto
@@ -8,9 +8,9 @@ package gnoi.packet_link_qualification;
 import "google/protobuf/duration.proto";
 import "google/protobuf/timestamp.proto";
 import "google/rpc/status.proto";
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
-option go_package = "github.com/openconfig/gnoi/linkqual";
+option go_package = "github.com/openconfig/gnoi/packet_link_qualification;linkqual";
 
 option (types.gnoi_version) = "1.1.0";
 
diff --git a/system/BUILD.bazel b/system/BUILD.bazel
index e1d0d09..5ac726e 100644
--- a/system/BUILD.bazel
+++ b/system/BUILD.bazel
@@ -29,6 +29,7 @@ package(
 proto_library(
     name = "system_proto",
     srcs = ["system.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = [
         "//common:common_proto",
         "//types:types_proto",
@@ -51,9 +52,15 @@ go_proto_library(
     name = "system_go_proto",
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/system",
-    protos = [":system_proto"],
+    proto = ":system_proto",
     deps = [
-        "//common:common_go_proto",
-        "//types:types_go_proto",
+        "//common",
+        "//types",
     ],
 )
+
+go_library(
+    name = "system",
+    embed = [":system_go_proto"],
+    importpath = "github.com/openconfig/gnoi/system",
+)
diff --git a/system/system.proto b/system/system.proto
index 5c5bda2..73c549d 100644
--- a/system/system.proto
+++ b/system/system.proto
@@ -24,8 +24,8 @@ syntax = "proto3";
 
 package gnoi.system;
 
-import "common/common.proto";
-import "types/types.proto";
+import "github.com/openconfig/gnoi/common/common.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/system";
 
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
index 04e4827..2c0c063 100644
--- a/test/BUILD.bazel
+++ b/test/BUILD.bazel
@@ -12,10 +12,10 @@ go_test(
     srcs = ["simple_test.go"],
     embed = [":test"],
     deps = [
-        "//bgp:bgp_go_proto",
-        "//common:common_go_proto",
-        "//system:system_go_proto",
-        "//types:types_go_proto",
+        "//bgp",
+        "//common",
+        "//system",
+        "//types",
         "@com_github_golang_protobuf//proto:go_default_library",
     ],
 )
diff --git a/types/BUILD.bazel b/types/BUILD.bazel
index 921d7c1..591bf29 100644
--- a/types/BUILD.bazel
+++ b/types/BUILD.bazel
@@ -29,6 +29,7 @@ package(
 proto_library(
     name = "types_proto",
     srcs = ["types.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = ["@com_google_protobuf//:descriptor_proto"],
 )
 
diff --git a/wavelength_router/BUILD.bazel b/wavelength_router/BUILD.bazel
index 67c0e19..3d33567 100644
--- a/wavelength_router/BUILD.bazel
+++ b/wavelength_router/BUILD.bazel
@@ -29,6 +29,7 @@ package(
 proto_library(
     name = "wavelength_router_proto",
     srcs = ["wavelength_router.proto"],
+    import_prefix = "github.com/openconfig/gnoi",
     deps = ["//types:types_proto"],
 )
 
@@ -49,5 +50,11 @@ go_proto_library(
     compilers = ["@io_bazel_rules_go//proto:go_grpc"],
     importpath = "github.com/openconfig/gnoi/wavelength_router",
     proto = ":wavelength_router_proto",
-    deps = ["//types:types_go_proto"],
+    deps = ["//types"],
+)
+
+go_library(
+    name = "wavelength_router",
+    embed = [":wavelength_router_go_proto"],
+    importpath = "github.com/openconfig/gnoi/wavelength_router",
 )
diff --git a/wavelength_router/wavelength_router.proto b/wavelength_router/wavelength_router.proto
index 1ec9bca..17989ae 100644
--- a/wavelength_router/wavelength_router.proto
+++ b/wavelength_router/wavelength_router.proto
@@ -18,7 +18,7 @@ syntax = "proto3";
 
 package gnoi.optical;
 
-import "types/types.proto";
+import "github.com/openconfig/gnoi/types/types.proto";
 
 option go_package = "github.com/openconfig/gnoi/wavelength_router";
 
