diff --git a/inc/sainexthop.h b/inc/sainexthop.h
index 5da33e9..47c38d7 100644
--- a/inc/sainexthop.h
+++ b/inc/sainexthop.h
@@ -50,6 +50,7 @@ typedef enum _sai_next_hop_type_t
     /** SRV6 SID List */
     SAI_NEXT_HOP_TYPE_SRV6_SIDLIST,
 
+
     /** IPMC next hop */
     SAI_NEXT_HOP_TYPE_IPMC,
 } sai_next_hop_type_t;
diff --git a/inc/sainexthopgroup.h b/inc/sainexthopgroup.h
index d591e24..03b40b1 100644
--- a/inc/sainexthopgroup.h
+++ b/inc/sainexthopgroup.h
@@ -59,6 +59,9 @@ typedef enum _sai_next_hop_group_type_t
     /** Next hop hardware protection group. This is the group backing up the primary in the protection group type and is managed by hardware */
     SAI_NEXT_HOP_GROUP_TYPE_HW_PROTECTION,
 
+    /** Next hop group is ECMP, with members specified with the group */
+    SAI_NEXT_HOP_GROUP_TYPE_ECMP_WITH_MEMBERS,
+
     /* Other types of next hop group to be defined in the future, e.g., WCMP */
 
 } sai_next_hop_group_type_t;
@@ -110,12 +113,16 @@ typedef enum _sai_next_hop_group_attr_t
     /**
      * @brief Next hop member list
      *
+     * Not valid when SAI_NEXT_HOP_GROUP_ATTR_TYPE == SAI_NEXT_HOP_GROUP_TYPE_ECMP_WITH_MEMBERS
+     *
      * @type sai_object_list_t
      * @flags READ_ONLY
      * @objects SAI_OBJECT_TYPE_NEXT_HOP_GROUP_MEMBER
      */
     SAI_NEXT_HOP_GROUP_ATTR_NEXT_HOP_MEMBER_LIST,
 
+
+
     /**
      * @brief Next hop group type
      *
@@ -235,6 +242,42 @@ typedef enum _sai_next_hop_group_attr_t
      */
     SAI_NEXT_HOP_GROUP_ATTR_ARS_PORT_REASSIGNMENTS,
 
+    /**
+     * @brief Next hop member list in the order specified by the application
+     *
+     * NH OID list length should match weight list length
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_NEXT_HOP
+     * @default empty
+     * @validonly SAI_NEXT_HOP_GROUP_ATTR_TYPE == SAI_NEXT_HOP_GROUP_TYPE_ECMP_WITH_MEMBERS
+     */
+    SAI_NEXT_HOP_GROUP_ATTR_NEXT_HOP_LIST,
+
+    /**
+     * @brief Next hop member weight list
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     * @validonly SAI_NEXT_HOP_GROUP_ATTR_TYPE == SAI_NEXT_HOP_GROUP_TYPE_ECMP_WITH_MEMBERS
+     */
+    SAI_NEXT_HOP_GROUP_ATTR_NEXT_HOP_MEMBER_WEIGHT_LIST,
+
+    /**
+     * @brief Next hop member counter list
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @default empty
+     * @validonly SAI_NEXT_HOP_GROUP_ATTR_TYPE == SAI_NEXT_HOP_GROUP_TYPE_ECMP_WITH_MEMBERS
+     */
+    SAI_NEXT_HOP_GROUP_ATTR_NEXT_HOP_MEMBER_COUNTER_LIST,
+
     /**
      * @brief End of attributes
      */
@@ -610,6 +653,11 @@ typedef struct _sai_next_hop_group_api_t
     sai_get_next_hop_group_map_attribute_fn    get_next_hop_group_map_attribute;
     sai_bulk_object_set_attribute_fn           set_next_hop_group_members_attribute;
     sai_bulk_object_get_attribute_fn           get_next_hop_group_members_attribute;
+    sai_bulk_object_create_fn                  create_next_hop_groups;
+    sai_bulk_object_remove_fn                  remove_next_hop_groups;
+    sai_bulk_object_set_attribute_fn           set_next_hop_groups_attribute;
+    sai_bulk_object_get_attribute_fn           get_next_hop_groups_attribute;
+
 } sai_next_hop_group_api_t;
 
 /**
diff --git a/inc/sairouterinterface.h b/inc/sairouterinterface.h
index bb52b6f..446fbd5 100644
--- a/inc/sairouterinterface.h
+++ b/inc/sairouterinterface.h
@@ -316,6 +316,15 @@ typedef enum _sai_router_interface_attr_t
      */
     SAI_ROUTER_INTERFACE_ATTR_SELECTIVE_COUNTER_LIST,
 
+    /**
+     * @brief Label attribute used to uniquely identify router interface.
+     *
+     * @type char
+     * @flags CREATE_AND_SET
+     * @default ""
+     */
+    SAI_ROUTER_INTERFACE_ATTR_LABEL,
+
     /**
      * @brief End of attributes
      */
diff --git a/inc/saitypes.h b/inc/saitypes.h
index b2a730a..618e551 100644
--- a/inc/saitypes.h
+++ b/inc/saitypes.h
@@ -106,6 +106,7 @@ typedef uint8_t sai_auth_key_t[16];
 typedef uint8_t sai_macsec_sak_t[32];
 typedef uint8_t sai_macsec_auth_key_t[16];
 typedef uint8_t sai_macsec_salt_t[12];
+typedef uint8_t sai_alpine_attr_t[64];
 
 #define _In_
 #define _Out_
@@ -1621,6 +1622,9 @@ typedef union _sai_attribute_value_t
     /** @validonly meta->attrvaluetype == SAI_ATTR_VALUE_TYPE_POE_PORT_POWER_CONSUMPTION */
     sai_poe_port_power_consumption_t portpowerconsumption;
 
+    /** @validonly meta->attrvaluetype == SAI_ATTR_VALUE_TYPE_ALPINE_ATTR */
+    sai_alpine_attr_t alpine_attr;
+
 } sai_attribute_value_t;
 
 /**
@@ -1799,6 +1803,29 @@ typedef struct _sai_stat_capability_list_t
 
 } sai_stat_capability_list_t;
 
+typedef struct _sai_stat_st_capability_t
+{
+    /**
+     * @brief Typical stat capability
+     */
+    sai_stat_capability_t capability;
+
+    /**
+     * @brief Minimal polling interval in nanoseconds
+     *
+     * If polling interval is less than this value, it will be unacceptable.
+     */
+    uint64_t minimal_polling_interval;
+
+} sai_stat_st_capability_t;
+
+typedef struct _sai_stat_st_capability_list_t
+{
+    uint32_t count;
+    sai_stat_st_capability_t *list;
+
+} sai_stat_st_capability_list_t;
+
 typedef enum _sai_stats_count_mode_t
 {
     /** Count packet and byte */
diff --git a/meta/saisanitycheck.c b/meta/saisanitycheck.c
index fb76035..c50f35a 100644
--- a/meta/saisanitycheck.c
+++ b/meta/saisanitycheck.c
@@ -5930,7 +5930,7 @@ void check_struct_and_union_size()
     CHECK_STRUCT_SIZE(sai_acl_action_parameter_t, 24);
     CHECK_STRUCT_SIZE(sai_acl_field_data_data_t, 16);
     CHECK_STRUCT_SIZE(sai_acl_field_data_mask_t, 16);
-    CHECK_STRUCT_SIZE(sai_attribute_value_t, 40);
+    CHECK_STRUCT_SIZE(sai_attribute_value_t, 64);
     CHECK_STRUCT_SIZE(sai_ip_addr_t, 16);
     CHECK_STRUCT_SIZE(sai_object_key_entry_t, 64);
     CHECK_STRUCT_SIZE(sai_tlv_entry_t, 36);
@@ -5945,7 +5945,7 @@ void check_struct_and_union_size()
     CHECK_STRUCT_SIZE(sai_acl_resource_list_t, 16);
     CHECK_STRUCT_SIZE(sai_acl_resource_t, 12);
     CHECK_STRUCT_SIZE(sai_attr_capability_t, 3);
-    CHECK_STRUCT_SIZE(sai_attribute_t, 48);
+    CHECK_STRUCT_SIZE(sai_attribute_t, 72);
     CHECK_STRUCT_SIZE(sai_bfd_session_state_notification_t, 16);
     CHECK_STRUCT_SIZE(sai_fabric_port_reachability_t, 8);
     CHECK_STRUCT_SIZE(sai_fdb_entry_t, 24);
